syncer/ddl.go

```diff
@@ -151,13 +153,38 @@ func (s *Syncer) splitAndFilterDDL(
 			}
 		}
 
-		shouldSkip, err2 := s.skipQuery(tableNames, stmt2, sql)
+		shouldSkip, isBaList, err2 := s.skipQuery(tableNames, stmt2, sql)
 		if err2 != nil {
 			return nil, nil, err2
 		}
 		if shouldSkip {
 			metrics.SkipBinlogDurationHistogram.WithLabelValues("query", s.cfg.Name, s.cfg.SourceID).Observe(time.Since(ec.startTime).Seconds())
-			ec.tctx.L().Warn("skip event", zap.String("event", "query"), zap.String("statement", sql), zap.String("schema", schema))
+			ec.tctx.L().Warn("skip event", zap.String("event", "query"), zap.String("sql", sql), zap.String("schema", schema), zap.Any("tableNames", tableNames))
+
+			if s.cfg.ShardMode == config.ShardOptimistic || isBaList || ec.currentLocation == nil {
```

Comment:
what's the reason we skip for `ShardOptimistic` and `currentLocation` 🤔

Reply:
`currentLocation` is to pass all unit test. The reason we skip `currentLocation` is that we all do not initialize `ec.currentLocation` in unit test.

Reply:
After skipping `isBaList(needTrack)`, I find we can dismiss `ShardOptimistic` check. I will remove it in next commit.